### PR TITLE
Criteo fixes and refactor

### PR DESCRIPTION
**Pull request by znado**
Subsuming https://github.com/mlcommons/algorithmic-efficiency/pull/122.

Fixing (hopefully) the criteo training issue, was a bug in shape broadcasting in the loss function.

Trains well, similar to many hparam settings internally. Still have not done a full run on the best hparams but is in progress.

### CODE REVIEW

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py [hunk 1]
```diff
@@ -165,37 +27,39 @@ def model_params_types(self):
...
   def loss_fn(
       self,
       label_batch: spec.Tensor,  # Dense (not one-hot) labels.
       logits_batch: spec.Tensor,
-      label_smoothing: float = 0.0,
-      mask_batch: Optional[spec.Tensor] = None) -> spec.Tensor:
-    smoothed_targets = optax.smooth_labels(label_batch, label_smoothing)
+      mask_batch: Optional[spec.Tensor] = None,
+      label_smoothing: float = 0.0) -> spec.Tensor:
+    del label_smoothing
```
Comment (runame) [inline algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:36]: Do we not want to be able to tune label smoothing for all workloads?

File: reference_submissions/criteo1tb/criteo1tb_jax/submission.py [hunk 2]
```diff
@@ -101,13 +101,14 @@ def update_params(workload: spec.Workload,
...
-  del global_step
+  # del global_step
 
   optimizer_state, opt_update_fn = optimizer_state
   per_device_rngs = jax.random.split(rng, jax.local_device_count())
-  new_model_state, new_optimizer_state, new_params = pmapped_train_step(
+  loss, new_model_state, new_optimizer_state, new_params = pmapped_train_step(
       workload, opt_update_fn, model_state, optimizer_state,
       current_param_container, batch, per_device_rngs)
+  print(f'Step {global_step} loss = {loss[0]}')
```
Comment (runame) [inline reference_submissions/criteo1tb/criteo1tb_jax/submission.py:111]: Should probably be deleted after debugging.

**Review by runame — APPROVED** [review]
LGTM!

File: algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py (hunk 1 above)
Comment (znado) [reply algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py:36]: We do, I just removed this here because for Criteo it's only binary classification, so I figured it would not be useful/valid to tune this, but I may have been mistaken? I added it to the agenda to chat about on Thurs to see if we wanna keep this.

File: reference_submissions/criteo1tb/criteo1tb_jax/submission.py (hunk 2 above)
Comment (znado) [reply reference_submissions/criteo1tb/criteo1tb_jax/submission.py:111]: lol thanks for catching this
